File: examples/PythonExamples-requests/GDBaaS/upload_from_external_file.py
# Copyright  2013, 2014, Ontotext AD
#
# This file is free software; you can redistribute it and/or modify it under
# the terms of the GNU Lesser General Public License as published by the Free
# Software Foundation; either version 2.1 of the License, or (at your option)
# any later version.
# This library is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
# details.
# You should have received a copy of the GNU Lesser General Public License
# along with this library; if not, write to the Free Software Foundation, Inc.,
# 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA

# IMPORTANT!!! Please install the packages below, needed for this example

# Warning! The following script uses information from example.rdf
# which is uploaded in the repo. Make sure you have both this file and
# example.rdf in the same directory before running this

import requests


endpoint = ("https://rdf.s4.ontotext.com/<user-id>/<database>/"
            + "repositories/<repo-id>/statements")
key = '<api-key>'
secret = '<api-secret>'
query = "Select * {?s ?p ?o} limit 50"

headers = {'Content-Type': 'application/rdf+xml;charset=UTF-8'}

with open('example.rdf', 'rb') as data_file:
    req = requests.post(
        endpoint, headers=headers, data=data_file, auth=(key, secret))

response = req.content.decode('utf-8')
print (response, '\n')

# Response status code
print ('Request Code: {}\n'.format(req.status_code))

# Response Headers
head = dict(req.headers)
print ('Headers: ')
for each in head:
    print (each.capitalize(), ": ", head[each])

File: examples/nodeJSExamples/GDBaaS/upload_from_external_file.js
/*
Copyright 2016 Ontotext AD

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

var fs= require("fs");
var https = require("https");

var headers = {
    "Content-Type": "application/rdf+xml;charset=UTF-8"
};

var options = {
    hostname: "rdf.s4.ontotext.com",
    path: "/<user-id>/<db>/repositories/<repository>/statements",
    method: "POST",
    auth: "<s4 api key>:<s4 key secret>",
    headers: headers
};

// Setup the request.  The options parameter is
// the object we defined above.
var req = https.request(options, function (res) {
    res.setEncoding("utf-8");

    var responseString = "";

    res.on("data", function (data) {
        responseString += data;
    });

    res.on("end", function () {
        console.log(responseString);
        console.log("\nStatus Code: ", res.statusCode);
        console.log("\nHeaders:");
        console.log(res.headers);
    });
});

req.on("error", function (e) {
    console.log(e);
});

var POST_DATA = fs.readFileSync("example.rdf").toString()
req.write(POST_DATA)

req.end();
